descqa/NumberDensityVersusRedshift.py

```diff
@@ -0,0 +1,307 @@
+
+        #other
+        self._other_kwargs = kwargs
+
+        return
+
+    def init_plots(self, mlo, mhi):
+        #get magnitude cuts based on validation data or default limits
+        mag_lo = self.validation_data.get('mag_lo', [float(m) for m in range(mhi, mlo+1)])
+        mag_hi = self.validation_data.get('mag_hi', [])
+
+        #setup plots and determine number of rows required for subplots
+        self.nplots = len(mag_lo)
+        self.nrows = (self.nplots+self.ncolumns-1)//self.ncolumns
+
+        #colors
+        self.colors = iter(default_colors)
+
+        return mag_lo, mag_hi
+
+    def get_catalog_data(self, gc, quantities, filters=[]):
+        
+        data = {}
+        if not gc.has_quantities(quantities):
+            return TestResult(skipped=True,summary='Missing requested quantities')
+        
+        data = gc.get_quantities(quantities,filters=filters)
+        #make sure data entries are all finite 
+        data = GCRQuery(*((np.isfinite, col) for col in data)).filter(data)
+
+        return data
+
+    def get_validation_data(self, band, observation):
+
+        validation_data = {}
+        z0errors = np.asarray([])
+        datafile = data_info[observation]['filename_template'].format(band)
+        filename = os.path.join(self.data_dir,datafile)
+        usecols = data_info[observation]['usecols']
+        skiprows= data_info[observation]['skiprows']
+
+        if os.path.exists(filename) and os.path.getsize(filename) > 0:
+            if observation.find(Coil2004)!=-1:
+                if len(usecols)==4:
+                    mag_hi,mag_lo,z0values,z0errors = np.loadtxt(filename,unpack=True,usecols=usecols,skiprows=skiprows)
+                    validation_data['z0errors'] = z0errors
+                else:
+                    mag_hi,mag_lo,z0values = np.loadtxt(filename,unpack=True,usecols=usecols,skiprows=skiprows)
+                validation_data['z0values'] = z0values
+                validation_data['mag_hi'] = mag_hi
+                validation_data['mag_lo'] = mag_lo
+                validation_data['label'] = Coil2004_lbl
+
+            elif observation.find(DEEP2_JAN)!=-1:
+                mag_hi_lim,mag_lo_lim,z0const,z0linear = np.loadtxt(filename,unpack=True,usecols=[0,1,2,3],skiprows=1)
+                validation_data['z0const'] = z0const
+                validation_data['z0linear'] = z0linear
+                validation_data['mag_hi'] = []
+                validation_data['mag_lo'] = [float(m) for m in range(int(mag_hi_lim),int(mag_lo_lim)+1)]
+                validation_data['label'] = DEEP2_JAN_lbl
+            else:
+                raise ValueError('Observation {} not available'.format(observation))
+        else:
+            raise ValueError("{}-band data file {} not found or is empty".format(band,filename))
+            
+        return validation_data
+
+    def run_on_single_catalog(self, galaxy_catalog, catalog_name, base_output_dir):
+        
+        #check for skip_messages
```

Comment:
Are lines 152-154 still needed? Remove them if not.

Reply:
Done